### ISSUE TITLE
~ tilde is not suppored in a global config

### ISSUE THREAD

**Issue by thebino** [rustic-rs/rustic#996]
If I want to use `~` in any path in the global config, it will not parse the config and stop with a non-helpful error.

For example if I want to write a logfile to the local .config directory
```
log-file = "~/.config/rustic/rustic.log"
```

Same for sources
```
[[backup.sources]]
source = "~/foo"
```

On mac the error message is also kind of strange (without showing which config is used)
```
error: rustic-rs fatal error: I/O operation failed: No such file or directory (os error 2)
```

**Comment by aawsome** [comment]
As `~` is IMO just a special case of variable substitution (here the env variable `$HOME`), I'm closing this as duplicate of #942

**Comment by simonsan** [comment]
Discovered this library lately: https://docs.rs/crate/shellexpand/3.1.0

Maybe worth another discussion, if we can implement it with that more easily?

Reopening for that discussion

**Comment by simonsan** [comment]
Copying over the discussion points of env var expansion:

This feature is a bit more involved as there are more questions arising:
- Do we only want variable substitution in filenames or elsewhere in the config?
- Do we want to allow substition only within strings?
- Do we want to first parse the TOML, then substitute what is appropriate or first substitute and then parse (which would allow to define TOML structures within env variables)?

_Originally posted by @aawsome in https://github.com/rustic-rs/rustic/issues/942#issuecomment-1887059882_

**Comment by Dialga** [comment]
ditto, as a first time user, i am also affected by this.
my config has `repo = '~/repo'` but instead `./~/repo` is used.

**Comment by rdelaage** [comment]
I didn't see this issue before, but I think it should be closed as #1577 brings environment variable substitution in config file. Users can use something like "$HOME/.config/rustic/rustic.log", or even "${XDG_CONFIG_HOME:$HOME/.config}/rustic/rustic.log"

**Comment by aawsome** [comment]
I agree with @rdelaage. With #1577, environment variable substitutions have been added allowing to use `$HOME` or `${HOME}` in the config profiles. I don't see the need to substitute also `~`.
